[REF] website_(sale_)slides: simplify _filter_add_members

flch-odoo opened the pull request:
An argument wasn't used, and the override wasn't necessary.
If we do need to be able to filter differently, based on access rights,
we should probably split the current into a proper overridable filter
and a check-access-and-raise-or-not-method, but it isn't necessary
at this point.

Task-3299646

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/149429).

std-odoo commented on addons/hr_skills_slides/__manifest__.py:
```diff
@@ -18,6 +18,7 @@
         'views/hr_employee_views.xml',
         'views/hr_templates.xml',
         'data/hr_resume_data.xml',
+        'data/mail_templates_chatter.xml',
```
can we add a test with `assertQueryCount` in a commit before this one, and update the count in this commit to see the improvement ?

std-odoo commented on addons/hr_skills_slides/models/slide_channel.py:
```diff
@@ -48,52 +50,47 @@ def _recompute_completion(self):
+        :param str kind: "joined", "left", "completed"
+        :param partners: res.partner recordset
+        """
+        if not partners or not self.ids or kind not in {"joined", "left", "completed"}:
```
```suggestion
        if not partners or not self or kind not in {"joined", "left", "completed"}:

std-odoo commented on addons/website_slides/models/slide_channel.py:
```diff
@@ -956,18 +956,20 @@ def _action_add_members(self, target_partners, member_status='joined', raise_on_
+        if controlled_access := self - allowed:
+            # If not raise, check each type independently as they may have different rules.
+            batches = [controlled_access] if raise_on_access else controlled_access.grouped('enroll').values()
+            for controlled_access_with_enroll in batches:
```
`check_access_rights` can be called once because it just depends on the model, will something like that work ?
```py
batches.check_access_rights('write')
allowed = batches._filter_access_rules('write')
if batches - allowed and raise_on_access:
    raise AccessError(_('You are not allowed to add members to this course. Please contact the course responsible or an administrator.'))
```
(assuming batches is a recordset and not a list)

std-odoo reviewed (commented):
@flch-odoo Hi :)  

Nice cleanup :) I have a few suggestions

Cheers

awa-odoo commented on addons/website_slides/models/slide_channel.py:
```diff
@@ -1002,14 +1002,13 @@ def _action_add_members(self, target_partners, member_status='joined', raise_on_
+        if controlled_access := (self - allowed):
+            allowed += controlled_access._filtered_access('write')
+            if raise_on_access and allowed != self:
+                raise AccessError(_('You are not allowed to add members to this course. '
```
Hello @flch-odoo 

As you know, we're trying to clean stalled tasks a bit.

Looking at this one, I think we could merge it, but maybe it would be nice to have a test with a "assertRaises" that makes sure this correctly fails when we want.

I checked and I did not see anything in the tests on "_action_add_members" with assertRaises so I don't think this is tested, maybe I did not look properly.

If you have an hour and want to do something else than documents, could be nice :disguised_face: 

Side-note: I don't think "CLN" is a proper commit tag, it's usually "REF" or "IMP", but I could be wrong.

Thanks & cheers!

awa-odoo replied:
@flch-odoo ? 🥲

awa-odoo replied:
@flch-odoo :pray:

flch-odoo commented:
Thanks for the review, @awa-odoo !

I've added a raise assertion just under the existing ones already making sure that the records were not created, and I've renamed the commit "type" as indeed CLN doesn't seem to exist (anymore? as I'm not the only one to use it, see below, (nothing special about the first few)).

Anyway, I've just fixed the style in this push, see the [comparison](https://github.com/odoo/odoo/compare/1f42bc5deda590d92d9f2c6e17e5466cee730dd9..e7349512f57ac61401f2a1b922fa3b8dd0bcb22e)

![image](https://github.com/user-attachments/assets/57da4cc3-11e6-40d4-9279-8fd77a630dda)

flch-odoo commented:
Ah, the docs doesn't specify it but CI checks and is still OK with it :-)

> Using regex ^([(FIX|REF|ADD|REM|REV|MOV|IMP|MERGE|CLA|I18N|REL|PERF|LINT|CLN)] .+) to check commit messages
